docs/rules/prefer-array-index-of.md

```diff
@@ -1,6 +1,6 @@
 # Prefer [`Array#indexOf()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) over [`Array#findIndex()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) when looking for the index of an item
 
-[`Array#findIndex()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) is made for more complex criteria. If you are just looking for the index where a given object is present, then the code can be reduced. It concerns any search with a literal, a variable or any expression that hasn't any side effect. However, if the expression you are looking for has side effect, relies on element related to the function (the name of the function, its arguments...), the case is still valid.
+[`Array#findIndex()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) is intended for more complex needs. If you are just looking for the index where the given item is present, then the code can be simplified. This applies to any search with a literal, a variable, or any expression that doesn't have any side effects. However, if the expression you are looking for has side effects, relies on element related to the function (the name of the function, its arguments, etc.), the case is still valid.
```

Comment:
>  However, if the expression you are looking for has side effects, relies on element related to the function (the name of the function, its arguments, etc.), the case is still valid.

This part seems not right, the searching value has "side effects"  not valid.

```js
const index = foo.findIndex(x => x === call());
```

This `call` has "side effects", but it will still reported without "auto-fix".

Reply:
I have added _**explicit** side effects_.

Reply:
In [`getBinaryExpressionSelector`](https://github.com/Fabrice-TIERCELIN/eslint-plugin-unicorn/blob/845d4727aea41e335ac227dbbdd4160de0a4f595/rules/prefer-array-index-of.js#L17), I want to add a criteria that avoids the existence of `call()`, something like:
```
:not(:has(*[expression.callee.name="call"][expression.type="CallExpression"]))
```

But I don't know how to tell to search only in the binary expression.